group/filippo_ed25519/scalar.go

```diff
@@ -0,0 +1,184 @@
+package filippo_ed25519
+
+import (
+	"crypto/cipher"
+	"encoding/hex"
+	"errors"
+	filippo_ed25519 "filippo.io/edwards25519"
+	"fmt"
+	"go.dedis.ch/kyber/v3"
+	"go.dedis.ch/kyber/v3/group/internal/marshalling"
+	"go.dedis.ch/kyber/v3/group/mod"
+	"go.dedis.ch/kyber/v3/util/random"
+	"io"
+	"math/big"
+)
+
+type Scalar struct {
+	scalar *filippo_ed25519.Scalar
+}
+
+func (s *Scalar) Equal(a kyber.Scalar) bool {
+	return s.scalar.Equal(a.(*Scalar).scalar) == 1
+}
+
+func (s *Scalar) Set(a kyber.Scalar) kyber.Scalar {
+	if s.scalar == nil {
+		s.scalar = new(filippo_ed25519.Scalar)
+	}
+	_ = s.scalar.Set(a.(*Scalar).scalar)
+	return s
+}
+
+func (s *Scalar) Clone() kyber.Scalar {
+	s2 := new(Scalar)
+	s2.scalar = new(filippo_ed25519.Scalar)
+	s2.scalar.Set(s.scalar)
+	return s2
+}
+
+func (s *Scalar) SetInt64(v int64) kyber.Scalar {
+	return s.setInt(mod.NewInt64(v, primeOrder))
+}
+
+func (s *Scalar) Zero() kyber.Scalar {
+	b := [32]byte{0}
+	if s.scalar == nil {
+		s.scalar = new(filippo_ed25519.Scalar)
+	}
+	_, err := s.scalar.SetCanonicalBytes(b[:])
+	if err != nil {
+		fmt.Println(err)
+	}
+	return s
+}
+
+func (s *Scalar) One() kyber.Scalar {
+	b := [32]byte{1}
+	if s.scalar == nil {
+		s.scalar = new(filippo_ed25519.Scalar)
+	}
+	_, err := s.scalar.SetCanonicalBytes(b[:])
```

Comment:
What is `SetCanonicalBytes` being used for here and elsewhere ? The (canonicalized) return value is ignored... 🤔